test/libsolidity/syntaxTests/inlineAssembly/circular_constant_access_err.sol

```diff
@@ -0,0 +1,11 @@
+contract C {
+    bytes32 constant x = x;
+    function f() public pure returns (uint t) {
+        assembly {
+            // Reference to a circular member
+            t := x
+        }
+    }
+}
+// ----
+// TypeError 3558: (171-172): Constant variable is circular.
```

Comment:
How are circular references handled elsewhere? Non inline assembly I mean.

Would it make sense to add the test case that came up in an earlier thread

```
    ==== Source: a ====
    import "b";
    uint constant c = d;
    ==== Source: b ====
    import "a" as M;
    uint constant b = M.c;
    uint constant d = b;
    contract C {
        uint constant a = b;
    }
```

Reply:
This already exists in `test/libsolidity/syntaxTests/constants/cross_file_cyclic_modules.sol`

Reply:
Perhaps add the following test case if it doesn't exist?

```
==== Source: a ====
bytes32 constant x = x;
==== Source: b ====
import "a";
contract C {
    function f() public pure returns (uint t) {
        assembly {
            // Reference to a circular member
            t := x
        }
    }
}
```